elfpy/data/contract_interface.py

```diff
@@ -56,15 +69,76 @@ class HyperdriveAddressesJson:
     mock_hyperdrive_math: str = attr.ib()
 
 
-def get_account_balance_for_contract(funding_contract: Contract, account_address: str) -> int:
+def initialize_web3_with_http_provider(ethereum_node: URI | str, request_kwargs: dict | None = None) -> Web3:
+    """Initialize a Web3 instance using an HTTP provider and inject a geth Proof of Authority (poa) middleware.
+
+    Arguments
+    ---------
+    ethereum_node: URI | str
+        Address of the http provider
+    request_kwargs: dict
+        The HTTPProvider uses the python requests library for making requests.
+        If you would like to modify how requests are made,
+        you can use the request_kwargs to do so.
+    """
+    if request_kwargs is None:
+        request_kwargs = {}
+    provider = Web3.HTTPProvider(ethereum_node, request_kwargs)
+    web3 = Web3(provider)
+    web3.middleware_onion.inject(geth_poa.geth_poa_middleware, layer=0)
+    return web3
+
+
+def set_account_balance(web3: Web3, account_address: str, amount_wei: int) -> RPCResponse:
+    """Set an the account using the web3 provider
+
+    Arguments
+    ---------
+    amount_wei : int
+        amount_wei to fund, in wei
+    """
+    # TODO: Assert that we are using anvil (devnet) to be able to call anvil_setBalance
+    params = [account_address, hex(amount_wei)]  # web3.to_hex(amount_wei)]  # account, amount
+    tx_receipt = web3.provider.make_request(method="anvil_setBalance", params=params)
+    return tx_receipt
+
+
+def fund_account(
+    web3: Web3, funding_contract: Contract, account_address: str, amount_wei: int, from_address: str
+) -> HexBytes:
+    """Add funds to the account
+
+    Arguments
+    ---------
+    amount_wei : int
+        amount_wei to fund, in wei
+    funding_contract : Contract | None
+        If a Contract, call the `mint` functionfor the amount.
+    """
+    tx_hash = funding_contract.functions.approve(account_address, amount_wei).transact(
```

Comment:
we don't need an approve call in `fund_account`.  we need approves after an account has the token and needs to authorize a smart contract to move some of those tokens.